Loop/Extensions/BasalRelativeDose.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  BasalRelativeDose.swift
+//  Loop
+//
+//  Created by Pete Schwamb on 2/12/24.
+//  Copyright © 2024 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+import LoopAlgorithm
+
+public extension Array where Element == BasalRelativeDose {
+    func trimmed(from start: Date? = nil, to end: Date? = nil, onlyTrimTempBasals: Bool = false) -> [BasalRelativeDose] {
+        return self.compactMap { (dose) -> BasalRelativeDose? in
+            if let start, dose.endDate < start {
+                return nil
+            }
+            if let end, dose.startDate > end {
+                return nil
+            }
+            if onlyTrimTempBasals && dose.type == .bolus {
+                return dose
+            }
```

Comment:
`BasalRelativeDose` is a type that holds doses (of either kind, basal or bolus) that are annotated with basal so their relative to basal impact can be calculated.  I did struggle with the name. `BasalAnnotatedDose` could be another option, I suppose.

The `onlyTrimTempBasals` flag indicates that only temp basals should be trimmed (split) if they overlap the boundaries of the time range. Looking at the code, I don't see any usage setting this flag to false (which makes sense to me).  I think I'll remove the flag.

Reply:
Another naming suggestion, since it is first a dose and then annotated by/relative to basal, `DoseBasalAnnotated` or similar. At least for me, I was confused since I thought this was some sort of basal dose.

But know I understand.